src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/TagHelperSemanticRangeVisitor.cs

```diff
@@ -386,8 +396,14 @@ private void AddSemanticRange(SyntaxNode node, int semanticKind)
             }
 
             var source = _razorCodeDocument.Source;
+
             var range = node.GetRange(source);
 
+            if (range.Start.Line != range.End.Line)
```

Comment:
I added this sanity check to prevent another multi-line structure from re-introducing this bug.

Reply:
How will this error get presented to users? Will it silently fail or gold bar? Also, why can't we produce multiple tokens when there's more thjen one line? Maybe I'm misunderstanding when this code actually gets hit but there's a good amount of multi-line support in Razor's syntax

Reply:
Hmm, I never realized it because I've literally never seen it done but you're correct that HTML allows for multi-line tokens in a couple different cases (HTML Attribute values being another). Given that I'll make this auto-split any multi-line token (though the API's on Node and Range aren't going to be particularly helpful in this.